src/_guides/language/effective-dart/usage_migrated.md

```diff
@@ -73,53 +73,98 @@ change to the package.
 That means that if you import some other package's private library, a minor,
 theoretically non-breaking point release of that package could break your code.
 
-### DO use relative paths when importing libraries within your own package's `lib` directory.
 
-{% include linter-rule.html rule1="avoid_relative_lib_imports" rule2="prefer_relative_imports" %}
+### DON'T allow an import path to reach into or out of `lib`.
 
-When referencing a library inside your package's `lib` directory from another
-library in that same package, use a relative URI, not an explicit `package:`
-URI.
+{% include linter-rule.html rule="avoid_relative_lib_imports" %}
+
+Dart supports `package:` imports to let you access
+the libraries inside a package's `lib` directory
+regardless of whether the package is physically stored on your computer.
+This can lead to a very confusing situation
+if you have a `package:` import of a library and another import
+that physically traverses into that same package's `lib` directory.
```

Comment:
I had trouble understanding this intro, even though I think I know what you're getting at. It seems a little too high level. Maybe:

```suggestion
If you have a `package:` import of a library and another import
that physically traverses into that same package's `lib` directory,
the result can be all sorts of errors and confusion.
```

Reply:
"physically traverses" could probably be simpler, now that I look at this. "physically" isn't really accurate, and "traverses" is not a word I'd expect an ESL reader to know.

Change it to "crosses"?

Reply:
Yeah, this is hard to explain concisely. I'll take another stab at it.